packages/journey-manager/src/queries.ts

```diff
@@ -126,12 +126,27 @@ export async function evaluate(q: Query): Promise<boolean> {
  * @param q - Query
  * @returns promise of a HTML element
  */
-export async function find(q: Query): Promise<HTMLElement> {
-  const container = q.parent != null ? await find(q.parent) : document.body;
+export async function find(q: Query): Promise<HTMLElement[]> {
+  const containers = q.parent != null ? await find(q.parent) : [document.body];
 
-  const methodName: keyof typeof queryFns = `findBy${q.queryName}`;
-
-  return await queryFns[methodName](container, ...(q.queryArgs as QueryArgs));
+  let results: HTMLElement[][] = [];
+  if (q.queryName === "QuerySelector") {
+    results = await Promise.all(
+      containers.map((container) => [
+        ...container.querySelectorAll<HTMLElement>(q.queryArgs[0] as string),
+      ]),
```

Comment:
```suggestion
       containers.map((container) =>
        waitFor(
          () => {
            const els = [
              ...container.querySelectorAll(q.queryArgs[0] as string),
            ];
            if (els.length == 0) {
              throw new Error("No elements found");
            }
          },
          { container },
        ),
      ),
```

where `waitFor` comes from `"@testing-library/dom"`

Reply:
@jakub-nlx how do you get away without returning `els`, am I missing something? This is what I could come up with:

```ts
      containers.map((container) =>
        waitFor(
          () => {
            const els = [
              ...container.querySelectorAll<HTMLElement>(
                q.queryArgs[0] as string,
              ),
            ];
            if (els.length == 0) {
              throw new Error("No elements found");
            }
            return els;
          },
          { container },
        ),

```

Reply:
Oh yeah, you definitely want to return `els` 🤦